web/src/components/Header/HeaderRightPanel.tsx

```diff
@@ -33,21 +35,57 @@ function StatusMenuItem({ color, title, onClick }) {
   )
 }
 
+function SearchResultItem({text, name}) {
+  return (
+    <div className="search-result-item-wrapper">
+      <Icon
+        name={name}
+        size="small"
+        className="light-grey not-hoverable"
+      />
+      <div className="search-result-item-text">{text}</div>
+    </div>
+  )
+}
+
+function SearchResultsFilter({name}) {
+  const [isFilterApplied, setIsFilterApplied] = useState(false)
+  return (
+    <div
+      className={`search-results-filter-wrapper ${
+        isFilterApplied ? 'filter-is-applied' : ''
+      } `}
+      onClick={() => {
+        setIsFilterApplied(!isFilterApplied)
+      }}
+    >
+      {name}
+    </div>
+  )
+}
+
 function HeaderRightPanel({
   hideGuidebookHelpMessage,
   whoami,
   onClickEditProfile,
   onClickPreferences,
   saveStatus,
   status,
+  goals,
+  goalComments,
 }) {
+
+  const goalList = Object.values(goals)
+  const commentList = Object.values(goalComments)
```

Comment:
You could move these Object.values calls to be inside the mapStateToProps function